src/index.js

```diff
@@ -26,7 +37,18 @@ function Block (data) {
       hashFunc = 'sha2-256'
     }
 
-    multihashing(this.data, hashFunc, callback)
+    if (this._cache[hashFunc]) {
+      setImmediate(() => {
```

Comment:
yeah we were shipping one before, but not anymore. As we already use `async` I'm going to use `async.setImmediate` http://caolan.github.io/async/docs.html#setImmediate